functions/Get-DbaTempdbUsage.ps1

```diff
@@ -0,0 +1,92 @@
+        {
+            try
+            {
+		        $server = Connect-SqlServer -SqlServer $SqlServ -SqlCredential $SqlCredential -RegularUser
+            }
+            catch
+            {
+                Write-Warning "Cannot connect to $SqlServ"
+                continue
+            }
+
+            if ($server.VersionMajor -le 9)
+            {
+                Write-Warning "This function is only supported in SQL Server 2008 or higher."
+                continue
+            }
+            
+            if ($Detailed -eq $true)
+            {
+                $sql = 'SELECT t.session_id AS spid, r.command AS statement_command, r.start_time, t.user_objects_alloc_page_count * 8 AS user_object_allocated_space, t.user_objects_dealloc_page_count * 8 AS user_object_deallocated_space, t.internal_objects_alloc_page_count * 8 AS internal_object_allocated_space, t.internal_objects_dealloc_page_count * 8 AS internal_object_deallocated_space, r.reads AS request_reads, r.writes AS request_writes, r.logical_reads AS request_logical_reads, r.cpu_time AS request_cpu_time, s.is_user_process, s.[status], DB_NAME(s.database_id) AS originating_database_name, s.login_name, s.original_login_name, s.nt_domain , s.nt_user_name, s.[host_name], s.[program_name], s.login_time, s.last_request_start_time, s.last_request_end_time FROM sys.dm_db_session_space_usage AS t INNER JOIN sys.dm_exec_sessions AS s ON s.session_id = t.session_id INNER JOIN sys.dm_exec_requests AS r ON r.session_id = s.session_id WHERE t.user_objects_alloc_page_count + t.user_objects_dealloc_page_count + t.internal_objects_alloc_page_count + t.internal_objects_dealloc_page_count > 0'
+                $datatable = $server.ConnectionContext.ExecuteWithResults($sql).Tables
+                return $datatable
+            }
+            else 
+            {
+                $sql = 'SELECT t.session_id AS spid, r.command AS statement_command, r.start_time, t.user_objects_alloc_page_count * 8 AS user_object_allocated_space, t.user_objects_dealloc_page_count * 8 AS user_object_deallocated_space, t.internal_objects_alloc_page_count * 8 AS internal_object_allocated_space, t.internal_objects_dealloc_page_count * 8 AS internal_object_deallocated_space, r.reads AS request_reads, r.writes AS request_writes, r.logical_reads AS request_logical_reads, r.cpu_time AS request_cpu_time, s.is_user_process, s.[status], DB_NAME(s.database_id) AS originating_database_name, s.login_name FROM sys.dm_db_session_space_usage AS t INNER JOIN sys.dm_exec_sessions AS s ON s.session_id = t.session_id INNER JOIN sys.dm_exec_requests AS r ON r.session_id = s.session_id WHERE t.user_objects_alloc_page_count + t.user_objects_dealloc_page_count + t.internal_objects_alloc_page_count + t.internal_objects_dealloc_page_count > 0;'
+                $datatable = $server.ConnectionContext.ExecuteWithResults($sql).Tables
+                return $datatable
+            }
+
+            $server.ConnectionContext.Disconnect()
```

Comment:
Your function is failing the command help validation:
 - You've still got your help parameter name set to Credential, rather than SqlCredential
 - You'll also need to provide help for the automatic parameters -WhatIf and -Confirm (these are automatic part of ShouldProcess, but don't get automatic help, which is one of those oddities we too often forget about)